extensions/ql-vscode/src/remote-queries/variant-analysis-view.ts

```diff
@@ -1,28 +1,212 @@
-import { ViewColumn } from 'vscode';
+import { ExtensionContext, ViewColumn } from 'vscode';
 import { AbstractWebview, WebviewPanelConfig } from '../abstract-webview';
-import { WebviewMessage } from '../interface-utils';
 import { logger } from '../logging';
+import { FromVariantAnalysisMessage, ToVariantAnalysisMessage } from '../pure/interface-types';
+import { assertNever } from '../pure/helpers-pure';
+import {
+  VariantAnalysis,
+  VariantAnalysisQueryLanguage,
+  VariantAnalysisRepoStatus,
+  VariantAnalysisStatus
+} from './shared/variant-analysis';
 
-export class VariantAnalysisView extends AbstractWebview<WebviewMessage, WebviewMessage> {
-  public openView() {
+export class VariantAnalysisView extends AbstractWebview<ToVariantAnalysisMessage, FromVariantAnalysisMessage> {
+  public static readonly viewType = 'codeQL.variantAnalysis';
+
+  public constructor(
+    ctx: ExtensionContext,
+    private readonly variantAnalysisId: number
+  ) {
+    super(ctx);
+  }
+
+  public async openView() {
     this.getPanel().reveal(undefined, true);
+
+    await this.waitForPanelLoaded();
   }
 
   protected getPanelConfig(): WebviewPanelConfig {
     return {
-      viewId: 'variantAnalysisView',
-      title: 'CodeQL Query Results',
+      viewId: VariantAnalysisView.viewType,
+      title: `CodeQL query results for query ${this.variantAnalysisId}`,
       viewColumn: ViewColumn.Active,
       preserveFocus: true,
-      view: 'variant-analysis'
+      view: 'variant-analysis',
     };
   }
 
   protected onPanelDispose(): void {
     // Nothing to dispose currently.
   }
 
-  protected async onMessage(msg: WebviewMessage): Promise<void> {
-    void logger.log('Received message on variant analysis view: ' + msg.t);
+  protected async onMessage(msg: FromVariantAnalysisMessage): Promise<void> {
+    switch (msg.t) {
+      case 'viewLoaded':
+        this.onWebViewLoaded();
+
+        void logger.log('Variant analysis view loaded');
+
+        await this.postMessage({
+          t: 'setVariantAnalysis',
+          variantAnalysis: this.getVariantAnalysis(),
+        });
+
+        break;
+      case 'stopVariantAnalysis':
```

Comment:
This is not yet used but is required for TypeScript, otherwise it will complain about the `assertNever` branch (since its a union of only 1 type, so equal to the 1 type)